OutOfSchool/OutOfSchool.DataAccess/Repository/ApplicationRepository.cs

```diff
@@ -50,6 +51,13 @@ public new async Task<Application> Update(Application entity)
 
             dbContext.Entry(application).State = EntityState.Modified;
 
+            if (application.Status != ApplicationStatus.Rejected)
```

Comment:
repository shouldn't contain logic. Need to check the status in controller or service